qubes/api/admin.py

```diff
@@ -1134,10 +1135,14 @@ async def _vm_create(self, vm_type, allow_pool=False,
             raise
         self.app.save()
 
-    @qubes.api.method('admin.vm.CreateDisposable', no_payload=True,
+    @qubes.api.method('admin.vm.CreateDisposable',
         scope='global', write=True)
-    async def create_disposable(self):
+    async def create_disposable(self, untrusted_payload):
         self.enforce(not self.arg)
+        if untrusted_payload not in (b"", b"uuid"):
```

Comment:
In the description you said it should be given in the argument, not the payload. So, which one is it?